api/src/test/java/org/apache/iceberg/transforms/TestTimestamps.java

```diff
@@ -241,4 +243,54 @@ public void testTimestampsReturnType() {
     Type hourResultType = hour.getResultType(type);
     assertThat(hourResultType).isEqualTo(Types.IntegerType.get());
   }
+
+  @Test
+  public void apply_bad_source_type() {
+    Timestamps badSourceType =
+        new Timestamps(ChronoUnit.CENTURIES, Timestamps.ResultTypeUnit.YEARS, "year");
+    assertThatThrownBy(() -> badSourceType.apply(11L))
+        .isInstanceOf(UnsupportedOperationException.class)
+        .hasMessageMatching("Unsupported source type unit: Centuries");
+  }
+
+  @Test
+  public void apply_bad_result_type() {
+    Timestamps badResultType =
+        new Timestamps(ChronoUnit.MICROS, Timestamps.ResultTypeUnit.NANOS, "nano");
+    assertThatThrownBy(() -> badResultType.apply(11L))
+        .isInstanceOf(UnsupportedOperationException.class)
+        .hasMessageMatching("Unsupported result type unit: NANOS");
+  }
+
+  @Test
+  public void get_TimestampType_ChronoUnit() {
+    Types.TimestampType timestampType = Types.TimestampType.withZone();
+    assertThatThrownBy(() -> Timestamps.get(timestampType, ChronoUnit.CENTURIES))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageMatching("Unsupported source/result type units: timestamptz->Centuries");
```

Comment:
nit: I think it would be good to have a space before/after `->`

Reply:
done